src/BSON/Document.c

```diff
@@ -229,6 +257,30 @@ static bool php_phongo_document_has(php_phongo_document_t* intern, char* key, si
 	return bson_iter_find_w_len(&iter, key, key_len);
 }
 
+static bool php_phongo_document_has_by_zval(php_phongo_document_t* intern, zval* key)
+{
+	if (Z_TYPE_P(key) != IS_STRING && Z_TYPE_P(key) != IS_LONG) {
+		return false;
+	}
+
+	zend_string* tmp_str;
+	zend_string* str = zval_try_get_tmp_string(key, &tmp_str);
```

Comment:
C99 allows declarations anywhere, but if there is a concern I can position these at the top of the function for consistency (granted, I don't think we have anything to enforce that moving forward since it's no longer a build error).